npc/re/instances/FridayDungeon.txt

```diff
@@ -0,0 +1,782 @@
+		enablenpc instance_npcname("#fd_box"+'chest3);
+		instance_announce -1, "A treasure chest appeared somewhere in the dungeon. It'll disappear after a while, so let's find it.",bc_map,"0xff5500";
+		sleep 180000;	//3min before chest disappears
+		disablenpc instance_npcname("#fd_box"+'chest3);
+	}
+	if ('ct == 400) {
+		'rand++;
+		if ('rand >4) 'rand = 1;
+		'chest4 = 'rand;	//remember in case someone kills 100 enemies within 3 minutes
+		enablenpc instance_npcname("#fd_box"+'chest4);
+		instance_announce -1, "A treasure chest appeared somewhere in the dungeon. It'll disappear after a while, so let's find it.",bc_map,"0xff5500";
+		sleep 180000;	//3min before chest disappears
+		disablenpc instance_npcname("#fd_box"+'chest4);
+	}
+	return;
+
+OnInstanceInit:
+	'ct = 0;
+	'freestones = 0;
+	'boss_actv = 0;
+	'map$ = instance_mapname(strnpcinfo(4));
+	end;
+}
+
+
+1@md_gef,110,129,6	script	Mary Juj#0_2	4_F_JOB_BLACKSMITH,{
+	npctalk "Hello";
+	end;
+OnInstanceInit:
+	disablenpc();
+	end;
+}
+
+1@md_gef,183,222,5	script	The Stranger#cadaver	4_TOWER_11,{
+	mes "It is a stranger's corpse.";
+	mes "Perhaps he died while wandering in this place on the search for treasure.";
+	next;
+	switch(select("Calm down:Investigate the body")) {
+	case 1:
+		mes "You respect the dead and decided to leave the body alone.";
+		close;
+	case 2:
+		mes "Looking at a hastily scribbled note in his pocket, perhaps the adventurer's name was Walter.";
+		next;
+		mes "[Adventurer's Log]";
+		mes "My name is Walter. For the adventurers who will visit this place in the future, I'm leaving this so that you don't make the same mistake again.";
+		next;
+		mes "[Adventurer's Log]";
+		mes "On the first day, I curiously put my hand on the slab of a strange Sculpture nearby.";
+		mes "...and I awokened something that I should not have.";
+		next;
+		mes "[Adventurer's Log]";
+		mes "Right now the revived ancient king is wandering in front of me.";
+		mes "Perhaps he has no memories of his previous life.";
+		mes "One thing is certain, he will kill me.";
+		next;
+		mes "[Adventurer's Log]";
+		mes "To the adventurers who will visit this place in the future, I hope you do not make the same mistake as me.";
+		mes "If you do see a suspicious slab, do not show any interest and pass by it.";
+		next;
+		mes "[Adventurer's Log]";
+		mes "Again, never touch the slab. Whatever temptation you may have.";
+		if('freestones==0) {
+			next;
+			mes "Several craft stones fall from his body. They were probably collected while exploring this place.";
+			close2;
+			if ( 'hard == 1 )
+				.@num = rand(6+getmapusers('map$),8+getmapusers('map$));	// Reward amount scales with player count
+			else
+				.@num = rand(4+getmapusers('map$),6+getmapusers('map$));	// Reward amount scales with player count
+				for ( .@i = 0; .@i < .@num; .@i++ )
+					makeitem 25235,1, 'map$, rand(181,183), rand(221,224);
+			'freestones=1;
```

Comment:
Anyone can talk to the NPC. 'the freestones variable must be rechecked in the case that several st are running
```suggestion
			if('freestones==0) {
				if ( 'hard == 1 )
					.@num = rand(6+getmapusers('map$),8+getmapusers('map$));	// Reward amount scales with player count
				else
					.@num = rand(4+getmapusers('map$),6+getmapusers('map$));	// Reward amount scales with player count
					for ( .@i = 0; .@i < .@num; .@i++ )
						makeitem 25235,1, 'map$, rand(181,183), rand(221,224);
				'freestones=1;
			}
```